build/bootstrap.sh

```diff
@@ -0,0 +1,220 @@
+    --install           # Which command to run, supported commands: builddependencies, developertools, containertools, qemu, all. Default is All
+EOM
+    exit 1
+}
+
+function parseArgs() {
+    arg_Positional=()
+    while [[ $# -gt 0 ]]; do
+        case $1 in
+        --help | -h)
+            usage
+            shift
+            exit 0
+            ;;
+        --platform | -p)
+            PLATFORM="${2,,}"
+            if [[ "$PLATFORM" == "debian9" || "$PLATFORM" == "debian10" || "$PLATFORM" == "ubuntu1804" || "$PLATFORM" == "ubuntu2004" ]];
+            then
+                echo -e "[INFO] Platform set to: ${PLATFORM}"
+            else 
+                echo -e "[ERROR] Unsupported platform: ${PLATFORM}"
+                exit
+            fi
+            
+            shift
+            ;;
+        --install | -i)
+            INSTALL="${2,,}"
+            echo -e "[INFO] Install command to run set to: ${INSTALL}"
+            shift
+            ;;
+        *)
+            arg_Positional+=("$1")
+            shift
+            ;;
+        esac
+    done
+}
+
+function installBuildDependencies
+{
+    if [[ -v PLATFORM ]]
+    then
+        echo "[INFO] Platform check succesful"
+    else
+        echo "[WARNING] No platform supplied, using default: Ubuntu1804"
+        PLATFORM = "ubuntu1804"
+    fi
+
+    echo "[INFO] Installing build dependencies"
+    apt-get install -y make build-essential g++ gdb gdbserver gcc git wget
+    apt-get install -y python3 ninja-build
+    
+    if [[ "$PLATFORM" == "debian9" ]];
+    then
+        # Cpprestsdk below requires min cmake version of 3.9, while 3.7 is the latest available on Debian9
+        # So build & install cmake from source
+        cd /tmp
+        wget https://cmake.org/files/v3.10/cmake-3.10.2.tar.gz
+        tar xzf cmake-3.10.2.tar.gz
+        cd /tmp/cmake-3.10.2
+        ./bootstrap
+        make
+        make install 
+        
+        # Install gsl from source, also not available on Debian9
+        cd /tmp/
+        git clone https://github.com/Microsoft/GSL.git
+        cd GSL/
+        git checkout tags/v2.0.0
+        cmake -DGSL_TEST=OFF .
+        make
+        make install 
+    else
+        apt-get -y install cmake libmsgsl-dev
+    fi
+
+    # Open-source library dependencies
+    # Boost libs for DO
+    apt-get install -y libboost-system-dev libboost-log-dev libboost-filesystem-dev libboost-program-options-dev
+    # Additional Boost libs for cpprestsdk
+    apt-get install -y libboost-random-dev libboost-regex-dev
+    apt-get install -y libproxy-dev libssl-dev uuid-dev
+
+    # Install cpprest dependencies
+    # libssl-dev also required but installed above because plugin uses libssl-dev directly
+    apt-get install -y zlib1g-dev
+
+    # Most target platforms do not natively have a version of cpprest that supports url-redirection
+    # Build and install v2.10.16 as it's the earliest version which supports url-redirection
+    mkdir /tmp/cpprestsdk
```

Comment:
>mkdir /tmp/cpprestsdk [](http://example.com/codeflow?start=4&length=21)

Script will fail if directory already exists. I've been bitten by this. Consider deleting the directory if exists and then continue with creating the dir.

Same for /tmp/gtest and other custom dirs we use.